# Задача 1

Вы провели эксперимент c упрощением формы заказа в магазине Утконос и получили результаты по метрике конверсий в покупку. Выберите метод оценки и оцените есть ли статистические значимые различия между конверсиями в двух группах при $\alpha = 5\%$. Дайте краткие рекомендации команде.

Результаты:
1. Число юзеров в группах , которые заходили на сайт в период эксперимента: $n_1 = 15550$ и $n_2 = 15550$.
2. Число юзеров в группах , которые совершили хотя бы одну покупку за период эксперимента: $n_1 = 164$ и $n_2 = 228$.
3. Конверсии: $conv_1 = 1.05\%$ и  $conv_2 = 1.47\%$.


## Решение

Считаем, что генеральная совокупность распределена нормально. Для оценки разницы конверсий используем Z-критерий и найдем доверительный интервал разницы, который определяется по формуле: $ \hat p_2 - \hat p_1 \pm Z_{\frac{\alpha}{2}} \cdot \sqrt{\frac{\hat p_1 \cdot (1 - \hat p_1)}{n_1} + \frac{\hat p_2 \cdot (1 - \hat p_2)}{n_2}}$, где $\hat p_1$ и $\hat p_2$ — значение конверсии 1 и 2 соответственно, $n_1$ и $n_2$ — размер выборки 1 и 2 соответственно, а $Z_{\frac{\alpha}{2}}$ — табличное критическое значение Z для выбранной $\alpha$ в двухстороннем тесте. При $\alpha = 0.05$ $Z_{\frac{\alpha}{2}} = 1.96$.

Сформулируем гипотезы:
* Нулевая гипотеза $H_0$ — между конверсиями в группах нет статистически значимой разницы
* Альтернативная гипотеза $H_1$ — между конверсиями в группах есть статистически значимая разница

Произведем расчеты доверительного интервала:

In [15]:
import numpy as np

n_1 = 15550
n_2 = 15550
sales_1 = 164
sales_2 = 228
hat_p_1 = sales_1 / n_1
hat_p_2 = sales_2 / n_2
z_score = 1.96
x_mean = hat_p_2 - hat_p_1
delta = z_score * np.sqrt(hat_p_1 * (1 - hat_p_1) /
                          n_1 + hat_p_2 * (1 - hat_p_2) / n_2)
print(f'[{round((x_mean - delta) * 100, 2)} %, {round((x_mean + delta) * 100, 2)} %]')


[0.16 %, 0.66 %]


Рассчитанный доверительный интервал [0.16 %, 0.66 %] не включает значение 0.0 при $\alpha = 0.05$, а значит нами выявлены статистически значимые различия между группами. Следовательно, гипотеза $H_0$ отвергается, а принимается гипотеза $H_1$.



## Ответ
По результатам проверки статистической значимости полученных результатов рекомендуется принять упрощенную форму заказов, так как она дает больше конверсий в покупку.

# Задача 2

Сравниваем метрику конверсия в покупку. Размер выборки - 10000 элементов в каждой группе. Какой статистический критерий тут лучше всего подойдёт и почему?

## Решение
Для оценки метрики конверсии могут быть использованы — $\chi$-квадрат на однородность или Z-критерий долей.
Z-критерий используется на больших числовых выборках из независимых групп распределение, при этом распредление должно быть нормальным.

Критерий $\chi$-квадрат используется, когда размер выборки невелик и для проверки гипотез о распределении категориальных переменных.

В рассматриваемом случае мы имеем большую выборку из двух независимых групп с нормальным распределением и сравниваем конверсии. Поэтому нужно использовать Z-критерий.

## Ответ
Z-критерий